commands/aliases.go

```diff
@@ -0,0 +1,203 @@
+	}
+
+	return nil
+}
+
+// writeToStorage overwrites the aliases held within `aliasMap` to the storage file.
+// Note that if an alias was removed from `aliasMap`, the removed alias will also
+// be removed from the storage file.
+func writeToStorage() error {
+	aliasBytes, err := json.Marshal(aliasMap)
+	if err != nil {
+		return fmt.Errorf("Failed to marshal JSON: %s", err)
+	}
+	if err := ioutil.WriteFile(aliasStorageFilepath, aliasBytes, 0744); err != nil {
+		return fmt.Errorf("Failed to write alias file: %s", err)
+	}
+	return nil
+}
+
+// withAliasCommands takes a map of command names to the command itself and
+// adds in a command where the command name is the alias key and the command
+// itself is a command which returns the alias value.
+func withAliasCommands(commMap map[string]Command) (map[string]Command, error) {
+	if err := loadFromStorage(); err != nil {
+		return nil, fmt.Errorf("failed to load aliases from storage: %s", err)
+	}
+
+	for aliasName, aliasValue := range aliasMap {
+		commMap[aliasName] = &textCommand{
+			helpText: aliasValue,
+			command: func(_ context.Context, _ []string) (string, error) {
+				return aliasValue, nil
+			},
+		}
+	}
+	return commMap, nil
+}
+
+// aliasCommand sets string => string shortcut that can be called later to print a value
+func aliasCommand(ctx context.Context, args []string) (string, error) {
+	l, loggerOk := logging.FromContext(ctx)
+	if loggerOk {
+		l.Infof("Responding to alias command")
+	}
+
+	if len(args) == 2 {
+		return "", errors.New("Too few arguments supplied. Refer to !help for usage")
+	}
+	if len(args) == 1 {
+		// shop a paginator of all aliases
+		p := dgwidgets.NewPaginator(
+			ctx.Value(messageContextValue("Session")).(*discordgo.Session),
+			ctx.Value(messageContextValue("ChannelID")).(string))
+		p.Add(&discordgo.MessageEmbed{
+			Color: 0,
+
+			Fields: []*discordgo.MessageEmbedField{
+				{
+					Name: "Aliases",
+					Value: func() string {
+						var sortedAliases []string
+						for a := range aliasMap {
+							sortedAliases = append(sortedAliases, fmt.Sprintf("**%s**", a))
+						}
+						sort.Strings(sortedAliases)
+						for i := range sortedAliases {
+							sortedAliases[i] = fmt.Sprintf("%d) %s", i+2, sortedAliases[i])
+						}
+						return strings.Join(sortedAliases, "\n")
+					}(),
+				},
+			},
+		})
+
+		for aliasName, aliasValue := range aliasMap {
+			embed := &discordgo.MessageEmbed{
+				Title:       aliasName,
+				Description: aliasValue,
+			}
+
+			resp, err := http.Head(aliasValue)
+			if err != nil {
+				p.Add(embed)
+				continue
+			}
+			defer resp.Body.Close()
+			content := strings.TrimPrefix(resp.Header.Get("Content-Type"), "image/")
+			if content == "gif" || content == "jpeg" || content == "png" {
+				embed.Image = &discordgo.MessageEmbedImage{
+					URL: aliasValue,
+				}
+			}
+			p.Add(embed)
+		}
+
+		p.SetPageFooters()
+		p.Loop = true
+		p.ColourWhenDone = 0xFF0000
+		p.DeleteReactionsWhenDone = true
+		p.Spawn()
```

Comment:
`p.Spawn()` returns an error, so we could do `return "", p.Spawn()`. I also feel like `p.Widget.Timeout = time.Minute * 5`. If i remember correctly, they never actually timed out and this may be the reason why

Reply:
Changed to 
```golang
p.SetPageFooters()
p.Loop = true
p.ColourWhenDone = 0xFF0000
p.DeleteReactionsWhenDone = true
p.Widget.Timeout = 2 * time.Minute
if err := p.Spawn(); err != nil {
	return "", fmt.Errorf("Failed to spawn paginator: %s", err)
}
return "", nil
```
The timeout will now be 2 mins. I wrapped the error so that it has context. Returning an error like that is rarely a good idea.